### PR TITLE
Fix #30876, make GITHUB_REGEX match ssh://git@github.com URLs.

### LINKED ISSUE #30876: ssh://git@github.com doesn't match LibGit2.GITHUB_REGEX
URLs of the form `ssh://git@github.com/tkoolen/PlanarConvexHulls.jl.git` (prevalent with `Pkg.setprotocol!("ssh")`) don't match `LibGit2.GITHUB_REGEX`, which e.g. causes issues with [`DocumenterTools.Travis.genkeys`](https://github.com/JuliaDocs/DocumenterTools.jl/blob/7206ae49592bff96d2cdc43c727e239a690d5e25/src/Travis.jl#L126).

### PR DESCRIPTION
**Pull request by tkoolen**
Fixes: #30876

### CODE REVIEW

File: stdlib/LibGit2/src/LibGit2.jl [hunk 1]
```diff
@@ -12,7 +12,7 @@ using Base.Printf: @printf
 export with, GitRepo, GitConfig
 
 const GITHUB_REGEX =
-    r"^(?:git@|git://|https://(?:[\w\.\+\-]+@)?)github.com[:/](([^/].+)/(.+?))(?:\.git)?$"i
+    r"^(?:git@|git://|https://|ssh://git@(?:[\w\.\+\-]+@)?)github.com[:/](([^/].+)/(.+?))(?:\.git)?$"i
```
Comment (omus) [inline stdlib/LibGit2/src/LibGit2.jl]: This change breaks HTTPS as the general username parsing needs to be applied only on that protocol. A test should be added along the lines of: "https://username@github.com"

Instead, we can do:
```suggestion
    r"^(?:(?:ssh://)?git@|git://|https://(?:[\w\.\+\-]+@)?)github.com[:/](([^/].+)/(.+?))(?:\.git)?$"i
```

File: stdlib/LibGit2/test/libgit2.jl [hunk 2]
```diff
@@ -570,6 +570,25 @@ end
         @test !LibGit2.ismatch("https://@github.com", cred)
         Base.shred!(cred)
     end
+
+    @testset "GITHUB_REGEX" begin
+        github_regex_test = function(url, user, repo)
+            m = match(LibGit2.GITHUB_REGEX, url)
+            @test m != nothing
```
Comment (omus) [inline stdlib/LibGit2/test/libgit2.jl]:
```suggestion
            @test m !== nothing
```

File: stdlib/LibGit2/test/libgit2.jl [hunk 3]
```diff
@@ -570,6 +570,25 @@ end
...
+        end
+        user = "User"
+        repo = "Repo"
+        github_regex_test("git@github.com/$user/$repo.git", user, repo)
+        github_regex_test("https://github.com/$user/$repo.git", user, repo)
+        github_regex_test("ssh://git@github.com/$user/$repo.git", user, repo)
+        github_regex_test("git@github.com/$user/$repo", user, repo)
+        github_regex_test("https://github.com/$user/$repo", user, repo)
+        github_regex_test("ssh://git@github.com/$user/$repo", user, repo)
+        @test match(LibGit2.GITHUB_REGEX, "git@notgithub.com/$user/$repo.git") == nothing
```
Comment (omus) [inline stdlib/LibGit2/test/libgit2.jl]:
```suggestion
        @test occursin(LibGit2.GITHUB_REGEX, "git@notgithub.com/$user/$repo.git")
```

**Review by omus — CHANGES_REQUESTED** [review]

File: stdlib/LibGit2/test/libgit2.jl (hunk 3 above)
Comment (tkoolen) [reply stdlib/LibGit2/test/libgit2.jl]: `!occursin`, I assume?

**Review by tkoolen — COMMENTED** [review]
Thanks for the comments.

File: stdlib/LibGit2/test/libgit2.jl (hunk 3 above)
Comment (omus) [reply stdlib/LibGit2/test/libgit2.jl]: Whoops, yes you are correct

**Review by omus — APPROVED** [review]

**Review by StefanKarpinski — APPROVED** [review]

**Comment by omus** [comment]
Looks like an unrelated network error

**Pull request merged**